BurnerB/FreeMentors-V2
# [#168006681] User Request Mentorship session

#### What does this PR do?
Allows a user to request a session from a mentor
#### Description of Task to be completed?
* Create session model and controller
* Create session validators
* Tests should pass
* Refractor tests
#### How should this be manually tested?
* Clone this  repository https://github.com/BurnerB/FreeMentors-V2.git and cd into FreeMentors-V2
* checkout to branch ft-request-session-168006681
* Run npm --init to install all dependancies.
* Run command npm test on your terminal
* Tests should pass
* Open postman and POST `/api/v1/sessions`
* Request spec:
```
{
	"mentorId":2,
	"questions":"Help me be a DJ man?"
}
```
 * Should recieve response:
 ```
 {
   "status": 200,
    "data":
        {
            "userId": 2,
            "firstName": "John",
            "lastName": "Doe",
            "email": "johndoe@email.com",
            "password": "$2b$10$T/IibLr1s1an27E7vG7kVuNxnDN4k/8ic1BgBOfHLDrJ59a5MPjHy",
            "address": "Nairobi Kenya",
            "bio": "rapper and record producer known as one of the most-controversial and best selling artists of the 21st century",
            "occupation": "Musician",
            "expertise": "rapping",
            "isMentor": true,
            "isAdmin": false
        },
}
 ```
#### What are the relevant pivotal tracker stories?
[#168006681](https://www.pivotaltracker.com/story/show/168006681)
#### Screenshots (if appropriate) 

![passing-request-session](https://user-images.githubusercontent.com/45785786/63579849-31a55680-c59c-11e9-9da3-c817e4a47872.png)
